File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        var maxContainerSize = 0;

        for (int i = 0; i < heights.length - 1; i++) {
            for (int j = i + 1; j < heights.length; j++) {
                var lowerBar = Integer.min(heights[i], heights[j]);
                var containerSize = lowerBar * (j - i);
                maxContainerSize = Integer.max(containerSize, maxContainerSize);
            }
        }

        return maxContainerSize;
    }
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        largest = float("-inf")
        
        for i in range(len(heights)):
            minHeight = float("inf")
            for j in range(i, len(heights)):
                minHeight = min(minHeight, heights[j])
                largest = max(largest, minHeight * (j - i + 1))

        return largest
        